flink-cdc-common/src/main/java/org/apache/flink/cdc/common/transform/converter/PostTransformConverter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.cdc.common.transform.converter;
+
+import org.apache.flink.cdc.common.annotation.Experimental;
+import org.apache.flink.cdc.common.event.DataChangeEvent;
+import org.apache.flink.cdc.common.utils.StringUtils;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * The PostTransformConverter applies to convert the {@link DataChangeEvent} after other part of
+ * TransformRule in PostTransformOperator.
+ */
+@Experimental
+public interface PostTransformConverter extends Serializable {
+
+    Optional<DataChangeEvent> convert(DataChangeEvent dataChangeEvent);
```

Comment:
Do we allow user to modify the before and after RecordData? if allow, we should provide Schema of this dataChangeEvent in this interface.

Reply:
This PR will only introduce SOFT_DELETE converter and I will revert some changes.